### PR TITLE
(#110) Begin replacing Assertion -> Assertion2

### PR DESCRIPTION
**Pull request by llorllale**
PR for #110:

* Replaced `Assertion` with `Assertion2` in about half of the test files
* Refactored some tests to use `Matches` where it made sense
* Fixed some of the messages in the assertions
* Added a couple of missing tests
* Left puzzle to continue work

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#111](https://github.com/llorllale/cactoos-matchers/issues/111) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#111](https://github.com/llorllale/cactoos-matchers/issues/111) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/f7bb950d-b278-4b6d-b825-ac642e37c3aa); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/llorllale/cactoos/matchers/HasLinesTest.java [hunk 1]
```diff
@@ -50,10 +50,10 @@
      */
```
Comment (ilyakharlamov) [inline src/test/java/org/llorllale/cactoos/matchers/HasLinesTest.java]: Do we still need 'PMD.AvoidDuplicateLiterals' here?

File: src/test/java/org/llorllale/cactoos/matchers/HasValuesMatchingTest.java [hunk 2]
```diff
@@ -47,10 +47,10 @@
      */
```
Comment (ilyakharlamov) [inline src/test/java/org/llorllale/cactoos/matchers/HasValuesMatchingTest.java]: why do we need 'PMD.AvoidDuplicateLiterals' here?

File: src/test/java/org/llorllale/cactoos/matchers/HasValuesMatchingTest.java [hunk 3]
```diff
@@ -47,10 +47,10 @@
...
     @Test
     public void matches() {
-        new Assertion<>(
-            "The matcher give the positive result for the valid arguments",
-            () -> new ListOf<>(1, 2, 3),
-            new HasValuesMatching<>(value -> value > 1 || value == 3)
+        new Assertion2<>(
+            "matches iterable with any elements that satisfy predicate",
+            new HasValuesMatching<>(value -> value > 1 || value == 3),
```
Comment (ilyakharlamov) [inline src/test/java/org/llorllale/cactoos/matchers/HasValuesMatchingTest.java]: `value > 1 || value == 3` ?
isn't 3 > 1 always?

File: src/test/java/org/llorllale/cactoos/matchers/IsBlankTest.java [hunk 4]
```diff
@@ -27,7 +27,6 @@
 
 package org.llorllale.cactoos.matchers;
 
-import org.hamcrest.core.StringContains;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
```
Comment (ilyakharlamov) [inline src/test/java/org/llorllale/cactoos/matchers/IsBlankTest.java]: Why 'PMD.AvoidDuplicateLiterals' here?

File: src/test/java/org/llorllale/cactoos/matchers/RunsInThreadsTest.java [hunk 5]
```diff
@@ -75,18 +73,18 @@ public void matchNegative() {
...
             )
         ).affirm();
-        new Assertion<>(
+        new Assertion2<>(
             "counter must not be incremented by all threads",
-            counter::get,
+            counter.get(),
             new IsNot<>(new IsEqual<>(threads * attempts))
         ).affirm();
     }
```
Comment (ilyakharlamov) [inline src/test/java/org/llorllale/cactoos/matchers/RunsInThreadsTest.java]: Exception 'java.lang.Exception' is never thrown in the methods here, why do we have `throws Exception` signature here?

Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/RunsInThreadsTest.java]: @ilyakharlamov I think you got confused here - there is no `throws Exception` in this test method

File: src/test/java/org/llorllale/cactoos/matchers/HasLinesTest.java (hunk 1 above)
Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/HasLinesTest.java]: @ilyakharlamov removed

File: src/test/java/org/llorllale/cactoos/matchers/HasValuesMatchingTest.java (hunk 2 above)
Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/HasValuesMatchingTest.java]: @ilyakharlamov removed

File: src/test/java/org/llorllale/cactoos/matchers/HasValuesMatchingTest.java (hunk 3 above)
Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/HasValuesMatchingTest.java]: @ilyakharlamov that was not the point, but I simplified it to `value -> value == 3` anyway

File: src/test/java/org/llorllale/cactoos/matchers/IsBlankTest.java (hunk 4 above)
Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/IsBlankTest.java]: @ilyakharlamov removed

File: src/test/java/org/llorllale/cactoos/matchers/RunsInThreadsTest.java (hunk 5 above)
Comment (ilyakharlamov) [reply src/test/java/org/llorllale/cactoos/matchers/RunsInThreadsTest.java]: @llorllale I meant `public Boolean apply(final AtomicInteger input) throws Exception {` no need for Exception here

Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/RunsInThreadsTest.java]: @ilyakharlamov fixed

**Comment by ilyakharlamov** [comment]
@llorllale looks good to me. 
Rebase so that history looks nice and clean and we are good to merge I think.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#111](https://github.com/llorllale/cactoos-matchers/issues/111) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request merged**